File: app/src/main/java/net/yupol/transmissionremote/app/BaseActivity.java
package net.yupol.transmissionremote.app;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onBackPressed() {
        boolean handled = handleBackPressByFragments();
        if (!handled) super.onBackPressed();
    }

    /**
     * @return {@code true} if back press handled by visible fragments
     */
    protected boolean handleBackPressByFragments() {
        for (Fragment fragment : getSupportFragmentManager().getFragments()) {
            if (fragment instanceof OnBackPressedListener && fragment.isVisible()) {
                boolean handled = ((OnBackPressedListener) fragment).onBackPressed();
                if (handled) return true;
            }
        }
        return false;
    }
}

File: app/src/main/java/net/yupol/transmissionremote/app/BaseActivity.kt
package net.yupol.transmissionremote.app

import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    override fun onBackPressed() {
        val handled = handleBackPressByFragments()
        if (!handled) super.onBackPressed()
    }

    /**
     * @return `true` if back press handled by visible fragments
     */
    protected fun handleBackPressByFragments(): Boolean {
        for (fragment in supportFragmentManager.fragments) {
            if (fragment is OnBackPressedListener && fragment.isVisible) {
                val handled = (fragment as OnBackPressedListener).onBackPressed()
                if (handled) return true
            }
        }
        return false
    }
}
